src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/services/WarningTaskHandlerTest.java

```diff
@@ -1,30 +1,127 @@
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
+@ExtendWith(MockitoExtension.class)
 class WarningTaskHandlerTest {
 
-    private final WarningTaskHandler handlerService = new WarningTaskHandler();
+    private static final String DMN_NAME = "wa-task-cancellation-ia-asylum";
+    public static final String WARN_TASKS_MESSAGE_NAME = "warnProcess";
+
+
+    @Mock
+    private WorkflowApiClientToWarnTask workflowApiClientToWarnTask;
+
+    @Captor
+    private ArgumentCaptor<SendMessageRequest<ProcessVariables, CancellationCorrelationKeys>> sendMessageRequestCaptor;
+
+    @InjectMocks
+    private WarningTaskHandler handlerService;
+
+    private final EventInformation eventInformation = EventInformation.builder()
+        .eventId("some event id")
+        .newStateId("some post state")
+        .previousStateId("some previous state")
+        .jurisdictionId("ia")
+        .caseTypeId("asylum")
+        .caseReference("some case reference")
+        .dateTime(LocalDateTime.now())
+        .build();
 
     @Test
     void evaluateDmn() {
-        assertThat(handlerService.evaluateDmn(
-            EventInformation.builder()
-                .jurisdictionId("ia")
-                .caseTypeId("asylum")
-                .build()))
-            .isEmpty();
+        CancellationEvaluateRequest cancellationTaskEvaluateDmnRequestVariables =
+            CancellationEvaluateRequest.builder()
+                .state(new DmnStringValue("some post state"))
+                .event(new DmnStringValue("some event id"))
+                .fromState(new DmnStringValue("some previous state"))
+                .build();
+
+        EvaluateDmnRequest<CancellationEvaluateRequest> requestParameters =
+            new EvaluateDmnRequest<>(cancellationTaskEvaluateDmnRequestVariables);
+
+        when(workflowApiClientToWarnTask.evaluateDmn(
+            DMN_NAME,
+            requestParameters
+        )).thenReturn(new EvaluateDmnResponse<>(Collections.emptyList()));
+
+        handlerService.evaluateDmn(eventInformation);
+
+        verify(workflowApiClientToWarnTask).evaluateDmn(
+            eq(DMN_NAME),
+            eq(requestParameters)
+        );
+
     }
 
+
     @Test
-    void handle() throws NoSuchMethodException {
-        // change for something meaningful once Warning handler is implemented
-        assertThat(handlerService.getClass().getMethod("evaluateDmn", EventInformation.class).getName())
-            .isEqualTo("evaluateDmn");
+    void handle() {
+        WarningEvaluateResponse result1 = WarningEvaluateResponse.builder()
+            .action(new DmnStringValue("Warn"))
+            .build();
+
+        WarningEvaluateResponse result2 = WarningEvaluateResponse.builder()
+            .action(new DmnStringValue("Warn"))
+            .build();
+
+        List<WarningEvaluateResponse> results = List.of(result1, result2);
```

Comment:
You can also have a test case where there is an empty list of results being sent.  It can check that then there is no interaction with the sendMessage endpoint.